pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -425,9 +447,79 @@ func (c *NPLController) handleAddUpdatePod(key string, obj interface{}) error {
 			}
 		}
 	}
-
 	if updatePodAnnotation {
-		return c.updatePodAnnotation(newPod)
+		return c.updatePodNPLAnnotation(pod, nplAnnotations)
+	}
+	return nil
+}
+
+// GetPodsAndGenRules fetches all the Pods on this Node and looks for valid NodePortLocal annotation,
+// if they exist with a valid Node Port, it adds the Node port to the port table and rules. If the Node port
+// is invalid or the NodePortLocal annotation is invalid, the NodePortLocal annotation is removed. The Pod
+// event handlers take care of allocating a new Node port if required.
+func (c *NPLController) GetPodsAndGenRules() error {
+	podList, err := c.kubeClient.CoreV1().Pods(metav1.NamespaceAll).List(context.TODO(), metav1.ListOptions{
```

Comment:
as pointed out earlier, you should use the `podLister` member of `NPLController` to list the Pods as it will used information which has already been fetched (and cached) from the K8s apiserver. There should be no need to do a new API request.

Reply:
Yeah, wasn't possible earlier because this func was getting called from `InitializeNPLAgent()`. Since we have now moved it to the controller's `Run()` function, I have changed it to use the `podLister`. Thanks.